azurerm/tags.go

```diff
@@ -56,6 +57,10 @@ func validateAzureRMTags(v interface{}, k string) (ws []string, es []error) {
 			es = append(es, fmt.Errorf("the maximum length for a tag key is 512 characters: %q is %d characters", k, len(k)))
 		}
 
+		if strings.EqualFold(k, "$type") {
+			es = append(es, fmt.Errorf("the %q is not allowed as tag name", k))
+		}
```

Comment:
given this method is shared across all tags (and this validation is only specific to the `metrics_alertrule` resource) - can we make a specific validation method for the that resource? we can call the main validation function, and then check that `$type` isn't included?

Reply:
Interesting, it seems that this file is inconsistent between local and remote. Let me check it and refresh it.

Reply:
I did some investigation for $type while using it in tag naming for metric_alertrule resource:

1. You can't directly use $type as naming of key in TF, you need to use "$type".
2. "$type" is not acceptable by API:
```
Error: Error applying plan:

1 error(s) occurred:

* azurerm_metric_alertrule.critical_cpu: 1 error(s) occurred:

* azurerm_metric_alertrule.critical_cpu: insights.AlertRulesClient#CreateOrUpdate: Failure responding to request: StatusCode=400 -- Original Error: autorest/azure: Service returned an error. Status=400 Code="UnsupportedRequestContent" Message="Request content is not well formed or supported."
```

Reply:
That's true for _this specific API_ but isn't the case across the whole of Azure, for instance I'm able to create a resource with `$type` as the key (note: this is double-escaped for HCL):

```
resource "azurerm_resource_group" "test" {
  name     = "tharvey-dev3"
  location = "West US"
  tags {
    "$$type" = "hi"
  }
}
```

in the case where this value isn't allowed we'll add validation logic to Terraform to the specific resources to handle this - but in general the key `$type` is a permitted value and as such we shouldn't obstruct that across the board.

Reply:
Please review my latest change, it seems that even if the PR is updated, the comment is still shown on old commit.